staging/src/k8s.io/apimachinery/pkg/util/cache/lruexpirecache.go

```diff
@@ -39,8 +38,11 @@ type LRUExpireCache struct {
 	// clock is used to obtain the current time
 	clock Clock
 
-	cache *lru.Cache
-	lock  sync.Mutex
+	lock sync.Mutex
+
+	maxSize      int
+	evictionList list.List
```

Comment:
list.New() returns `*List`... should this be a pointer?

Reply:
According to the docs, the zero value of List is an empty list that's ready to use.  It looks like it gets lazily initialized as elements are added.

Reply:
I saw that, but was confused how a non-pointer receiver would self-modify

Reply:
List's functions all take pointer receivers. Won't they just get the address of evictionList?  I admit, I'm a bit fuzzy on the actual behavior here.

Reply:
yeah, go converts the non-pointer receiver to a pointer automatically